### PR TITLE
spack list : merged package-list into list

### PR DESCRIPTION
**Pull request by alalazo**
50% of #1907

### CODE REVIEW

File: lib/spack/spack/cmd/list.py [hunk 1]
```diff
@@ -40,18 +50,27 @@ def setup_parser(subparser):
         help='Optional glob patterns to filter results.')
     subparser.add_argument(
         '-s', '--sensitive', action='store_true', default=False,
-        help='Use case-sensitive filtering. Default is case sensitive, '
+        help='Use case-sensitive filtering. Default is case insensitive, '
         'unless the query contains a capital letter.')
```
Comment (adamjstewart) [inline lib/spack/spack/cmd/list.py]: I just tested this out and apparently it is actually case sensitive. Try running:
```
$ spack list imagemagick
```
Honestly, there are so few packages in Spack with capital letters that I think we should always be case insensitive. We might just want to remove the option altogether.

Comment (alalazo) [inline lib/spack/spack/cmd/list.py]: @adamjstewart @tgamblin What would you like me to do here ? Personally I would remove the option and go always case insensitive....

Comment (adamjstewart) [inline lib/spack/spack/cmd/list.py]: There's so few packages with capital letters anyway:
```
$ spack list | grep [A-Z]
ImageMagick
libNBC
LuaJIT
Mitos
py-SQLAlchemy
R
r-BiocGenerics
r-R6
SAMRAI
```

**Comment by adamjstewart** [comment]
@alalazo Can you update the documentation to reflect these changes? The [documentation on `spack list`](http://spack.readthedocs.io/en/latest/basic_usage.html#cmd-spack-list) doesn't reflect the changes to case sensitivity.

**Comment by alalazo** [comment]
@adamjstewart Will do

**Pull request merged**